src/utils/domUtils.ts

```diff
@@ -64,7 +64,7 @@ export module Utils {
       return selection;
     }
 
-    export function boxesOverlap(boxA: ClientRect, boxB: ClientRect) {
+    export function clientRectOverlap(boxA: ClientRect, boxB: ClientRect) {
```

Comment:
I went for this approach since I was thinking "does clientRect 1 overlap clientRect 2?".  That said, your proposal works just as well.

What do you think?